SETUP/upgrade/20/20240205_alter_queue_defns.php

```diff
@@ -0,0 +1,106 @@
+<?php
+$relPath = '../../../pinc/';
+include_once($relPath.'base.inc');
+
+header('Content-type: text/plain');
+
+// ------------------------------------------------------------
+
+echo "Adding projects_target and pages_target to queue_defns...\n";
+$sql = "
+    ALTER TABLE queue_defns
+    ADD COLUMN projects_target SMALLINT NOT NULL AFTER release_criterion,
+    ADD COLUMN pages_target MEDIUMINT NOT NULL AFTER projects_target
+";
+mysqli_query(DPDatabase::get_connection(), $sql) or die(mysqli_error(DPDatabase::get_connection()));
+
+$sql = "
+    SELECT id, round_id, name, release_criterion
+    FROM queue_defns
+    ORDER BY id
+";
+$result = mysqli_query(DPDatabase::get_connection(), $sql) or die(mysqli_error(DPDatabase::get_connection()));
+$n_rows = mysqli_num_rows($result);
+echo "Converting the release_criterion in {$n_rows} rows ...\n";
+
+$n_conversion_errors = 0;
```

Comment:
I don't want to over-engineer the conversion script, but putting on my "roll this out to PROD" hat it would be nice to know about the errors before the database alterations happened. That way I'm notified about them, can go fix those, and then re-run the script rather than go update the pages/projects column for the failures and then drop the column.

Reply:
Done.